src/acceptance-test/java/de/zalando/aruha/nakadi/repository/db/EventTypeCacheTest.java

```diff
@@ -0,0 +1,154 @@
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.fail;
+import static org.echocat.jomon.runtime.concurrent.Retryer.executeWithRetry;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class EventTypeCacheTest {
+
+    private final EventTypeRepository dbRepo = mock(EventTypeRepository.class);
+    private final CuratorFramework client;
+
+    public EventTypeCacheTest() throws Exception {
+        String connectString = "127.0.0.1:2181";
+        RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 3);
+        CuratorFramework cf = CuratorFrameworkFactory.newClient(connectString, retryPolicy);
+        cf.start();
+        this.client = cf;
+    }
+
+    @Before
+    @After
+    public void setUp() throws Exception {
+        if (client.checkExists().forPath("/nakadi/event_types/event-name") != null) {
+            client.delete().forPath("/nakadi/event_types/event-name");
+        }
+    }
+
+    @Test
+    public void onCreatedAddNewChildrenZNode() throws Exception {
+        EventTypeCache etc = new EventTypeCache(dbRepo, client);
+
+        EventType et = buildEventType();
+
+        etc.created(et);
+
+        assertNotNull(client.checkExists().forPath("/nakadi/event_types/" + et.getName()));
+    }
+
+    @Test
+    public void whenUpdatedSetChildrenZNodeValue() throws Exception {
+        EventTypeCache etc = new EventTypeCache(dbRepo, client);
+
+        EventType et = buildEventType();
+
+        client
+                .create()
+                .creatingParentsIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/nakadi/event_types/" + et.getName(), "some-value".getBytes());
+
+        etc.updated(et.getName());
+
+        byte data[] = client.getData().forPath("/nakadi/event_types/" + et.getName());
+        assertThat(data, equalTo(new byte[0]));
+    }
+
+    @Test
+    public void whenRemovedThenDeleteZNodeValue() throws Exception {
+        EventTypeCache etc = new EventTypeCache(dbRepo, client);
+
+        EventType et = buildEventType();
+
+        client
+                .create()
+                .creatingParentsIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/nakadi/event_types/" + et.getName());
+
+        etc.removed(et.getName());
+
+        assertNull(client.checkExists().forPath("/nakadi/event_types/" + et.getName()));
+    }
+
+    @Test
+    public void loadsFromDbOnCacheMissTest() throws Exception {
+        EventTypeCache etc = new EventTypeCache(dbRepo, client);
+
+        EventType et = buildEventType();
+
+        Mockito
+                .doReturn(et)
+                .when(dbRepo)
+                .findByName(et.getName());
+
+        etc.get(et.getName());
```

Comment:
I think here you can also check the cache actually returned correct event type (first time and second time)